### PR TITLE
Implement a minitest runner for easier integration

### PR DESCRIPTION
**Pull request by casperisfine**
Depends on https://github.com/Shopify/ci-queue/pull/34

Here's what I got so far:

```
$ RUBYLIB=lib/ exe/minitest-queue -Itest:lib --url /tmp/test_order.log  --retry test/fixtures/dum*_test.rb 
Started with run options --seed 53139

ATest
  test_foo                                                        SKIP (0.00s)
  test_flaky                                                      PASS (0.00s)
  test_bar                                                        PASS (0.00s)

BTest
  test_bar                                                        PASS (0.00s)
  test_foo                                                        PASS (0.00s)

Finished in 0.00089s
5 tests, 1 assertions, 0 failures, 0 errors, 1 skips
Coverage report generated for Unit Tests to ~/src/github.com/Shopify/ci-queue/ruby/coverage. 182 / 398 LOC (45.73%) covered.
```

### CODE REVIEW

**Comment by casperisfine** [comment]
cc @Shopify/pipeline. I'm not 100% sure it's ready for review. I'd like to try integrate it more into core to see how it goes first. 

But if you are curious and have a bit of time feel free to comment. Otherwise I'll ping back later next week for an actual review.

File: ruby/lib/minitest/queue/runner.rb [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+          paths.split(':').reverse.each do |path|
+            $LOAD_PATH.unshift(File.expand_path(path))
+          end
+        end
+
+        @argv.each do |f|
+          require File.expand_path(f)
+        end
+
+        Minitest.queue.populate(shuffle(Minitest.loaded_tests), &:to_s) # TODO: stop serializing
```
Comment (DazWorrall) [inline ruby/lib/minitest/queue/runner.rb]: Is there ever a situation where we wouldn't want to shuffle here?

File: ruby/lib/ci/queue/file.rb [hunk 2]
```diff
@@ -3,8 +3,14 @@
...
-      def initialize(path, **args)
-        super(::File.readlines(path).map(&:strip).reject(&:empty?), **args)
+      class << self
+        def from_uri(uri, config)
+          new(uri.path, config)
+        end
+      end
+
+      def initialize(path, *args)
+        super(::File.readlines(path).map(&:strip).reject(&:empty?), *args)
```
[the remark this answers is not in the document]
Comment (casperisfine) [reply ruby/lib/ci/queue/file.rb]: Yes. Also you can perfectly catch keyword args with `*args` in ruby.

File: ruby/lib/minitest/queue/runner.rb (hunk 1 above)
Comment (casperisfine) [reply ruby/lib/minitest/queue/runner.rb]: Maybe, but no legitimate one. I'd rather force shuffling on everybody.

**Pull request merged**